app/controllers/admin/api/services/proxy/policies_controller.rb

```diff
@@ -17,7 +17,7 @@ def update
       ApicastV2DeploymentService.new(@proxy).call(environment: :sandbox)
     end
 
-    respond_with(proxy.policies_config, represent_on_error: :resource)
+    respond_with(proxy.policies_config, represent_on_error: represent_on_error)
```

Comment:
A brief explanation: In a proxy policy chain, the errors can be present at two levels:

- On or more of the policies contain an error (e.g. lacking the `version` field).
- The policies are right, but the whole chain is wrong (e.g. is too long, or no default police present)

When the whole chain is wrong, we want to return `422` along with something like this:

```
{
	"errors": {
		"policies_config": [
			"is too long (maximum is 16 characters)"
		]
	}
}
```
This way, 3scale_toolbox is able to understand it. Also, as errors are present for the whole chain, UI also detects the errors.

When is one of the policies which is wrong, we want to return `422`, but the response format is different:

```
{
	"policies_config": [
		{
			"name": "apicast",
			"configuration": {},
			"enabled": true,
			"errors": {
				"version": [
					"can't be blank"
				]
			}
		}
	]
}
```

That way the client can know which particular policy is wrong. Nor 3scale_toolbox or the UI use this, and this makes things more complicated, but since is already live we can't just remove it, we don't know if anybody is using it.

The `represent_on_error` option to `respond_with` is hack introduced by https://github.com/3scale/system/pull/9066 which forces to render the resource even when there are errors. The resource representer is modified to show the errors if any, [here](https://github.com/3scale/porta/blob/25fa9bc53bdd3eeb8f2a9c8e882f771c85c3c3cb/app/representers/policy_config_representer.rb#L10).

Reply:
I think we need to add a test for this functionality.

Reply:
Done!

Reply:
Great, thanks!